examples/systemcalls/systemcalls.c

```diff
@@ -58,7 +72,47 @@ bool do_exec(int count, ...)
  *   as second argument to the execv() command.
  *
 */
+    pid_t child_pid;
+
+    child_pid = fork();
+
+    if (child_pid == -1) { // Error in process creation
+        perror("Fork");
+        return false;
+    }
+    else if (child_pid == 0) { // Child Process
+        int ret;
+        ret = execv(command[0], command);
+
+        if (ret == -1) {
+            perror("execv");
+        }
+        printf("Error executing command: execv with error: %d \n", errno);
+        exit(1);
+    } else {
+        int status;
+        pid_t wait_pid = waitpid(child_pid, &status, 0);
+
+        if (wait_pid == -1) {
+            printf("Error executing command: waitpid with error: %d \n", errno);
+            perror("wait");
+            return false;
+        } else {
+            // Check if process exited normally
+            if (WIFEXITED (status)) {
+                // Check the exit status of the process
+               if (WEXITSTATUS(status) == 1) {
+                return false;
+               } else if (WEXITSTATUS(status) == 0) {
```

Comment:
@GautamaGandhi here you handle exit status 1 and 0 but what about all other status? I think your logic should include a handler for 0 as success and all others as failure here.

Reply:
@dwalkes I checked if the process exited normally using WIFEXITED (status). The else case for this indicates an error and returns false. The handler then checks for the exact exit status based on what is set in the child process (0 if success and 1 if failure).

Reply:
> The handler then checks for the exact exit status based on what is set in the child process (0 if success and 1 if failure).

I don't think you can guarantee that the child process will return 1 on failure, you just know it won't return 0.